src/systems/material.js

```diff
@@ -146,7 +146,7 @@ module.exports.System = registerSystem('material', {
     setTextureProperties(texture, data);
 
     // if we're on iOS, and the video is HLS, we currently need to do some hacks
-    if (utils.device.isIOS() && isHLS(videoEl)) {
+    if (utils.device.isIOS() && isHLS(videoEl.src || videoEl.getAttribute('src'), videoEl.type || videoEl.getAttribute('type'))) {
```

Comment:
here the check for the property and if none the getAttribute value was unexpectedly required for both synthetic tests and actual real-world usage to work correctly.